src/openms/include/OpenMS/FILTERING/DATAREDUCTION/IsotopeDistributionCache.h

```diff
@@ -47,16 +48,37 @@ namespace OpenMS
 public:
     typedef FeatureFinderAlgorithmPickedHelperStructs::TheoreticalIsotopePattern TheoreticalIsotopePattern;
 
-    IsotopeDistributionCache(double max_mass, double mass_window_width, double intensity_percentage = 0, double intensity_percentage_optional = 0);
+    /// @name Constructors and Destructors
+    //@{
+    /** Default constructor
+     */
+    IsotopeDistributionCache();
+
+    /// Destructor
+    ~IsotopeDistributionCache() = default;
+    //@}
+
+
+    void precalculateDistributionCache(Size num_begin, Size index);
+
+    void renormalize( TheoreticalIsotopePattern& isotopes, IsotopeDistribution& isotope_dist);
 
     /// Returns the isotope distribution for a certain mass window
-    const TheoreticalIsotopePattern & getIsotopeDistribution(double mass) const;
+    const TheoreticalIsotopePattern& getIsotopeDistribution(double mass) ;
+
+    const IsotopeDistribution& getIntensity(double mass);
```

Comment:
this is not really an intensity which is returned here, right?
Can you find a better name for the method?
Also document it